flytekit/core/python_auto_container.py

```diff
@@ -212,7 +212,7 @@ def get_k8s_pod(self, settings: SerializationSettings) -> _task_model.K8sPod:
         if self.pod_template is None:
             return None
         return _task_model.K8sPod(
-            pod_spec=_serialize_pod_spec(self.pod_template, self._get_container(settings)),
+            pod_spec=_serialize_pod_spec(self.pod_template, self._get_container(settings), settings),
```

Comment:
could we update `_get_container` instead of adding settings here?

Reply:
we could use `get_registerable_container_image(..)` [here](https://github.com/flyteorg/flytekit/blob/3cb79885049f9de6552d00bff1e9a57ba0d3ee0a/flytekit/core/container_task.py#L116)

Reply:
_get_container only get the primary container, but there might be many containers in a pod template. e.g.

```
image_spec_1 = ImageSpec(
    name="image-1",
    packages=["numpy"],
    registry="localhost:30000",
)

image_spec_2 = ImageSpec(
    name="image-2",
    packages=["pandas"],
    registry="localhost:30000",
)


ps = V1PodSpec(
    containers=[
        V1Container(
            name="primary",
            image=image_spec_1,
        ),
        V1Container(
            name="secondary",
            image=image_spec_2,
            # use 1 cpu and 1Gi mem
            resources=V1ResourceRequirements(
                requests={"cpu": "1", "memory": "100Mi"},
                limits={"cpu": "1", "memory": "100Mi"},
            )
        )
    ]
)

pt = PodTemplate(pod_spec=ps, primary_container_name="primary")



@task(
    pod_template=pt,
)
def t1():
    ...
```